var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -552,7 +552,13 @@ def test(self):
             runexe = Executable(join_path(self.prefix,
                                           'lib/petsc/bin/petsc-mpiexec.uni')).command
             runopt = ['-n', '1']
-        w_dir = join_path(self.install_test_root, 'src/ksp/ksp/tutorials')
+        w_dir = join_path(self.test_suite.current_test_cache_dir,
+                          'src', 'ksp', 'ksp', 'tutorials')
+
+        if not os.path.exists(w_dir):
+            print('Skipping petsc test')
+            return
```

Comment:
Perhaps @adamjstewart can verify if this is the correct way to add this test.

And how to update this - in case of multiple cached dirs as in https://github.com/spack/spack/pull/27624

Reply:
@tldahlgren is the expert on package testing

Reply:
This PR is a simple substitution for the existing test to ensure the test is being executed in the test cache directory (i.e., under `self.test_suite.current_test_cache_dir`) instead of under the package's install prefix (i.e., `self.install_test_root`) to avoid permissions issues related to shared instances and facility installations.

@balay Is the concern the hacky way of flagging the test is being skipped?  (This is a known issue we plan to address.) Or is it that you might want to add more tests in the future?  (In this case, one option is to have a separate method for each if they are from different source subdirectories.)

Note: #27624 could be passing both subdirectories in a list so there is only one call to the cache method that will copy files from both.

Reply:
@balay This PR wasn't intended to add a test but does add the current practice for logging that a test is being skipped due to missing files.  Would you prefer that the check be removed so the example fails in another way?

Reply:
Nvm.  I just revisited emails and noticed there were more differences in this PR last week.

Reply:
@tldahlgren 

> This PR is a simple substitution for the existing directory to ensure the test is being executed in the test cache directory

Thanks for this fix!

> Is the concern the hacky way of flagging the test is being skipped?

The concern was: the referred to PR has 2 dirs to be staged in. So how would the "skip test" be updated for that setup?
- check each dir - and skip that part of the test? [is that ok to do?] or
-  test both (all) locations - and either run all tests [or skip all tests]?

> Note: #27624 could be passing both subdirectories in a list 

Perhaps this fix can be added to this PR. #27624 is already merged in - so this branch needs to be rebased over (or merged in with) latest spack/develop.

 [and then add in this this PR change to the second location aswell.

Reply:
Excellent points.

@RikkiButler20 Please rebase this PR to pick up 27624 and adjust the changes here accordingly.  Feel free to contact me if you want to discuss it further.